pkg/operator/controllerimagepruner.go

```diff
@@ -0,0 +1,423 @@
+			klog.Errorf("unable to calculate difference in %s: %s", utilObjectInfo(pcr), err)
+		}
+		klog.Infof("object changed: %s (metadata=%t, spec=%t): %s", utilObjectInfo(pcr), metadataChanged, specChanged, difference)
+
+		updatedPCR, err := c.clients.RegOp.ImageregistryV1().ImagePruners().Update(pcr)
+		if err != nil {
+			if !errors.IsConflict(err) {
+				klog.Errorf("unable to update %s: %s", utilObjectInfo(pcr), err)
+			}
+			return err
+		}
+
+		// If we updated the Status field too, we'll make one more call and we
+		// want it to succeed.
+		pcr.ResourceVersion = updatedPCR.ResourceVersion
+
+	}
+
+	pcr.Status.ObservedGeneration = pcr.Generation
+	statusChanged := !reflect.DeepEqual(prevPCR.Status, pcr.Status)
+	if statusChanged {
+		difference, err := object.DiffString(prevPCR, pcr)
+		if err != nil {
+			klog.Errorf("unable to calculate difference in %s: %s", utilObjectInfo(pcr), err)
+		}
+		klog.Infof("object changed: %s (status=%t): %s", utilObjectInfo(pcr), statusChanged, difference)
+
+		_, err = c.clients.RegOp.ImageregistryV1().ImagePruners().UpdateStatus(pcr)
+		if err != nil {
+			if !errors.IsConflict(err) {
+				klog.Errorf("unable to update status %s: %s", utilObjectInfo(pcr), err)
+			}
+			return err
+		}
+	}
+
+	if _, ok := applyError.(permanentError); !ok {
+		return applyError
+	}
+
+	return nil
+}
+
+func (c *ImagePrunerController) eventProcessor() {
+	for {
+		obj, shutdown := c.workqueue.Get()
+		if shutdown {
+			return
+		}
+
+		klog.V(1).Infof("get event from image pruner workqueue")
+		func() {
+			defer c.workqueue.Done(obj)
+
+			if _, ok := obj.(string); !ok {
+				c.workqueue.Forget(obj)
+				klog.Errorf("expected string in workqueue but got %#v", obj)
+				return
+			}
+
+			if err := c.sync(); err != nil {
+				c.workqueue.AddRateLimited(imagePrunerWorkQueueKey)
+				klog.Errorf("(imaeg pruner) unable to sync: %s, requeuing", err)
+			} else {
+				c.workqueue.Forget(obj)
+				klog.Infof("event from image pruner workqueue successfully processed")
+			}
+		}()
+	}
+}
+
+func (c *ImagePrunerController) handler() cache.ResourceEventHandlerFuncs {
+	return cache.ResourceEventHandlerFuncs{
+		AddFunc: func(o interface{}) {
+			klog.V(1).Infof("add event to image pruner workqueue due to %s (add)", utilObjectInfo(o))
+			c.workqueue.Add(imagePrunerWorkQueueKey)
+		},
+		UpdateFunc: func(o, n interface{}) {
+			newAccessor, err := kmeta.Accessor(n)
+			if err != nil {
+				klog.Errorf("unable to get accessor for new object: %s", err)
+				return
+			}
+			oldAccessor, err := kmeta.Accessor(o)
+			if err != nil {
+				klog.Errorf("unable to get accessor for old object: %s", err)
+				return
+			}
+			if newAccessor.GetResourceVersion() == oldAccessor.GetResourceVersion() {
+				// Periodic resync will send update events for all known resources.
+				// Two different versions of the same resource will always have different RVs.
+				return
+			}
+			if clusterOperator, ok := o.(*configapiv1.ClusterOperator); ok {
```

Comment:
ClusterOperators are not watched by this controller

Reply:
thanks, thought i got all of those.